app/api/routers/openneuro.py

```diff
@@ -16,16 +18,43 @@
     SuccessfulUploadWithWarnings,
 )
 
-# TODO: Error out when PAT doesn't exist in environment
+# TODO: Error out when these variables are not set
 GH_PAT = os.environ.get("GH_PAT")
+API_USERNAME = bytes(os.environ.get("API_USERNAME"), encoding="utf-8")
+API_PASSWORD = bytes(os.environ.get("API_PASSWORD"), encoding="utf-8")
 
 router = APIRouter(prefix="/openneuro", tags=["openneuro"])
 
+security = HTTPBasic()
 
+
+def verify_credentials(credentials: HTTPBasicCredentials = Depends(security)):
```

Comment:
Yes, both should serve the same purpose - dependency injection - but just using different syntax for type hinting/declaring metadata about the parameter. On the page you linked, I'm simply using the ["Python 3.8+ non-Annotated"](https://fastapi.tiangolo.com/advanced/security/http-basic-auth/?h=http+basic#__tabbed_2_3) syntax, for consistency with our other APIs!

I think w/ newer versions of Python + FastAPI, the `Annotated` syntax (which is part of python's standard library now) is recommended. We may want to consider switching over at some point, but currently the FastAPI version used across our APIs is also restricted by the fact that we haven't upgraded to Pydantic 2+ yet...so that would probably be higher priority 😛

I'm not too familiar with using `Annotated` yet, but there's some good info on it [here](https://fastapi.tiangolo.com/python-types/#type-hints-with-metadata-annotations)!